src/components/look/look-list/style-tag-list/index.tsx

```diff
@@ -21,11 +21,14 @@ function StyleTagList({ styleTags, id }: StyleTagListProps) {
 
   const handleCheck = (id: number) => {
     const isSelected = selectedTagIds.includes(id);
-    setSelectedTagIds(
-      selectedTagIds
-        .filter((selectedTagId) => selectedTagId != id)
-        .concat(isSelected ? [] : id)
-    );
+    const isSelectable = isSelected || selectedTagIds.length < 2;
+    isSelectable
+      ? setSelectedTagIds(
+          selectedTagIds
+            .filter((selectedTagId) => selectedTagId != id)
+            .concat(isSelected ? [] : id)
+        )
+      : message.warning('스타일 태그를 3개 이상 선택할 수 없습니다.');
```

Comment:
Escape case를 먼저 처리하는게 로직 흐름상 좋습니다!

```suggestion
    const newSelectedTagIds = selectedTagIds
            .filter((selectedTagId) => selectedTagId != id)
            .concat(isSelected ? [] : id);

    if(newSelectedTagIds.length > 2) {
      message.warning('스타일 태그는 2개까지만 선택할 수 있습니다.');
      return;
    }
    
    setSelectedTagIds(newSelectedTagIds);
```

Reply:
9b0e455